src/init/chainstate.cpp

```diff
@@ -137,7 +137,7 @@ std::optional<ChainstateLoadingError> LoadChainstateSequence(bool fReset,
                 }
 
                 const CBlockIndex* tip = chainstate->m_chain.Tip();
-                if (tip && tip->nTime > GetAdjustedTime() + 2 * 60 * 60) {
+                if (tip && tip->nTime > get_unix_time_seconds() + 2 * 60 * 60) {
```

Comment:
In commit "init/chainstate: Decouple from GetAdjustedTime"

I suppose we can't just pass a timeout time in here because the current logic includes the runtime of the loading/checking itself?

This seems like yet something else that would be unnecessary if this function were split. That keeps coming up as a potential alternative, but I'm unsure if there's something that would prevent it, so I'll stop suggesting it.